kythe/go/platform/kcd/kythe/units.go

```diff
@@ -82,6 +83,48 @@ func (u Unit) Index() kcd.Index {
 	return idx
 }
 
+// ResolveVName satisfies part of the kcd.Unit interface.
+func (u Unit) ResolveVName(inputPath string) *spb.VName {
+	root := u.Proto.GetWorkingDirectory()
+	if root == "" {
+		root = "/"
+	}
+
+	// match inputPath against the cleaned, absolute path.
+	if path.IsAbs(inputPath) {
+		inputPath = path.Clean(inputPath)
+	} else {
+		inputPath = path.Join(root, inputPath)
+	}
+	for _, ri := range u.Proto.RequiredInput {
+		if ri.Info == nil {
+			continue
+		}
+		qp := ri.Info.GetPath()
+		if path.IsAbs(qp) {
+			qp = path.Clean(qp)
+		} else {
+			qp = path.Join(root, qp)
+		}
+		if qp == inputPath {
+			v := ri.GetVName()
```

Comment:
Either clone the proto or create a new one manually.

Reply:
Done. Also refactored this a bit.